test/Microsoft.TemplateEngine.Authoring.TemplateVerifier.IntegrationTests/TemplateEngineSamplesTest.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.TemplateEngine.Authoring.TemplateApiVerifier;
+using Microsoft.TemplateEngine.TestHelper;
+using Microsoft.TemplateEngine.Tests;
+using Xunit.Abstractions;
+
+namespace Microsoft.TemplateEngine.Authoring.TemplateVerifier.IntegrationTests
+{
+    public class TemplateEngineSamplesTest : TestBase
+    {
+        private readonly ILogger _log;
+
+        public TemplateEngineSamplesTest(ITestOutputHelper log)
+        {
+            _log = new XunitLoggerProvider(log).CreateLogger("TestRun");
+        }
+
+        [Theory]
+        [InlineData("01-basic-template", "sample01", null, "no args")]
+        [InlineData("02-add-parameters", "sample02", new[] { "copyrightName", "Test Copyright", "title", "Test Title" }, "text args")]
+        [InlineData("03-optional-page", "sample03", new[] { "enableContactPage", "true" }, "optional content included")]
+        [InlineData("03-optional-page", "sample03", null, "optional content excluded")]
+        [InlineData("04-parameter-from-list", "sample04", new[] { "BackgroundColor", "dimgray" }, "the parameter from the list of options")]
+        [InlineData("05-multi-project", "sample05", new[] { "--includetest", "true" }, "the optional test project included")]
+        [InlineData("05-multi-project", "sample05", new[] { "--includetest", "false" }, "the optional test project excluded")]
+        [InlineData("06-console-csharp-fsharp", "sample06", new[] { "--language", "F#" }, "the F# lang parameter that creates a corresponded project")]
+        [InlineData("06-console-csharp-fsharp", "sample06", new[] { "--language", "C#" }, "the C# lang parameter that creates a corresponded project")]
```

Comment:
Language parameter only works in CLI, so passing it does nothing.  @JanKrivanek am I right here? is there a way to diff the language in API tests?

Reply:
Yeah - unfortunately this is CLI functionality - so it would need to be replicated in the testing aPI extension - however it would be at least controversial if not boldly wrong. As we'd then test some functionality that's now in sdk, but we'd actualy test it's replicated version that's part of testing framework.

tl;dr; scenarios that need advanced CLI functionality should not be tested within templating repo, but rather sdk repo. Here (in templating), we need to explictly call each template separately

Reply:
I remove sample06 from here.
It will be covered in sdk

Reply:
@JanKrivanek i think it would be good addition to API verifier though.
Use case: the package contains the templates for same language:
|Identity|short-name|language|
|---|---|---|
|MyTemplate.FSharp|mytemplate| F#|
|MyTemplate.CSharp|mytemplate| C#|

There is no way to execute specific one from that one. Probably extending [here](https://github.com/dotnet/templating/blob/bb19dc09e1c32c1fd75f3539dc75590c5a08e028/tools/Microsoft.TemplateEngine.Authoring.TemplateApiVerifier/TemplateVerifierOptionsExtensions.cs#L98) to allow passing custom filter may work.

Reply:
Added this to https://github.com/dotnet/templating/issues/5705